### PR TITLE
Initial changes to create a `unit!` macro.

### PR DESCRIPTION
**Pull request by iliekturtles**
Additional changes still pending to allow the `quantity!` macro to also
be called outside the mod where the `system!` macro was called.

### CODE REVIEW

File: examples/unit.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+//! Example showing how to use the `unit!` macro to add new units to existing quantities.
+
+#[macro_use]
+extern crate uom;
```
Comment (adamreichold) [inline examples/unit.rs:4]: Maybe using the macro directly like
```
use uom::unit;
```
now that we are using edition 2018?

File: src/unit.rs [hunk 2]
```diff
@@ -0,0 +1,257 @@
+/// Macro to implement a set of [measurement units][measurement]. Note that this macro must be
+/// executed in direct submodules of the module where the [`system!`](macro.system.html) macro was
+/// executed. `@...` match arms are considered private.
+///
+/// * `$unit`: Unit name (e.g. `meter`, `foot`).
+/// * `$conversion`: Conversion (coefficient and constant factor) from the unit to the base unit of
```
Comment (adamreichold) [inline src/unit.rs:6]: The wording of "constant factor" did puzzle me a bit during the first reading  "factor" suggests a product where a sum is computed. AFAIU, this is something like an offset or an intercept?

Comment (iliekturtles) [reply src/unit.rs:6]: Most units just use a coefficient. The constant factor is for thermodynamic temperature. Thoughts on better wording?

https://github.com/iliekturtles/uom/blob/1a3386a7f999c6da5a366366b094028848795ed2/src/si/thermodynamic_temperature.rs#L90-L93

**Comment by iliekturtles** [comment]
Latest changes now ready for review.

I ended up stopping short of making the quantity! macro be able to be executed outside of direct submodules of the location where the system! macro was executed. Executing quantity! in an external crate runs into issues with attempting to implement external traits for external types and needing access to private items.

**Comment by iliekturtles** [comment]
Note to self: Update the documentation to state that PRs for any new units are desired.

**Pull request merged**